pxteditor/code-validation/getNestedChildBlocks.ts

```diff
@@ -0,0 +1,10 @@
+/// <reference path="../../localtypings/validatorPlan.d.ts" />
+
+import * as Blockly from "blockly";
+
+export function getNestedChildBlocks(parentBlock: Blockly.Block): Blockly.Block[] {
+    // want param to be false to filter out disabled blocks
+    const descendants = parentBlock.getDescendants(false);
```

Comment:
From the docs, I thought this flag was just an ordering parameter.

https://developers.google.com/blockly/reference/js/blockly.block_class.getdescendants_1_method

Even if it does have a side effect of removing disabled blocks, I don't think we should rely on undocumented behavior. I'd still recommend adding a check for disabled inside the filter condition.

Reply:
As a side note, idk if it's even possible for a child to be disabled when the parent is not disabled, so maybe a check on the parent block is sufficient? But that may require a bit more investigation.

Reply:
You're right, it is. I have no idea why I thought it filtered disabled blocks. I thought I read it somewhere, but I can't find it lol; maybe I dreamt it. 😆

Reply:
> As a side note, idk if it's even possible for a child to be disabled when the parent is not disabled, so maybe a check on the parent block is sufficient? But that may require a bit more investigation.

I agree; my first impression would be that it wouldn't be possible to have disabled child blocks if the parent is not, but I'm also not sure. 

On another note, I don't think we need to filter out disabled blocks here. All of the validators filter out disabled blocks (except comment validators), so we'd be doing extra work. We can take out the filtering in the different validators and put it in `runValidatorPlan`, but if we do that, the tutorial validation might get different results from the teacher tool validation.

Reply:
And, I'm going to change this from false to true. I think ordered children will generally be more helpful if we want to take advantage of the ordered capability.

Reply:
As a helper function, I don't think we can make assumptions about what the caller will/won't have done regarding disabled blocks, however you could add a parameter filterDisabled and validators could pass false if they think it's unnecessary. That said, I _think_ the check is pretty trivial so I wouldn't expect any significant impact by running it.